src/components/settings-menu/PrivacySection.tsx

```diff
@@ -19,6 +25,20 @@ const PrivacySection = () => {
     publicShowCase => !publicShowCase,
     webDataEnabled
   );
+  const [analyticsEnabled, toggleAnalytics] = useReducer(analyticsEnabled => {
+    device.set(['doNotTrack'], analyticsEnabled);
+    if (analyticsEnabled) {
+      analyticsV2.track(analyticsV2.event.analyticsTrackingDisabled);
+    } else {
+      analyticsV2.track(analyticsV2.event.analyticsTrackingEnabled);
+    }
```

Comment:
Dig the reducer!

The `analyticsV2` instance already emits these events when calling enable/disable. Tbh though, here is probably better. Esp once we add those methods to the logger too, makes sense to enable/disable in one place, and emit these events in once place.

So let's remove those events from the analytics instance (will need to update tests quick too) and keep them here. That can happen in this PR.

Reply:
Good call setting the local storage first. That way if anything else fails, at least after the next app start their selection should be respected.